qsv: fix format selection, segmentation fault, proper naming

galinart opened the pull request:
[UPD]: Remove colorspace converter from QSV encoder  
[UPD]: disabled unsupported profiles and level for AV1 encoder 

Basically it is a fix of regression only. There is no new functionality here.

* Remove colorspace converter from QSV encoder into nv12 or p010 formats in case of system memory. 
It was broken when "format filter" was introduced and no need to insert it in pipeline in case of SW decoder and QSV encoder. Otherwise double conversion happens and leads to segmentation fault.

galinart committed:
qsv: fix segmentation fault due to colorspace conversions, remove conversion from encoder

Co-authored-by: Vladyslav Sosunovych <vladyslav.sosunovych@intel.com>

galinart committed:
qsv: proper naming for filters variables

bradleysepos commented:
Cool, the renaming definitely helps understandability.

galad87 commented:
Could this be changed to use the format filter to avoid code duplication or is there something special in the way the qsv encoder uses swscale?

Please align the * to the variable name.

galinart committed:
qsv: allow to select only supported av1 profiles and levels

galinart commented:
> Could this be changed to use the format filter to avoid code duplication or is there something special in the way the qsv encoder uses swscale?

Historically it was there, but now pipeline changed to support other formats. Agree, this is one of our next steps. It will require some changes in encoder logic. We will do it iteratively, for now fixed this regression in Nightly build. 

> 
> Please align the * to the variable name.

Sorry, * fixed.

sr55 commented:
Initial testing looks good here. Thanks Artem!

galinart commented:
@galad87 remove conversion from encoder as you suggested

galad87 commented:
As usual I can't test, but the code looks good.

sr55 commented:
I'll give it a pass later on this evening and get it in if it all looks good.
